prov/efa/test/efa_unit_test_hmem.c

```diff
@@ -0,0 +1,51 @@
+#include "efa_unit_tests.h"
+
+
+#if HAVE_NEURON
+/**
+ * @brief Verify when neuron_alloc failed (return null),
+ * efa_domain_open, which call efa_hmem_support_status_update_neuron
+ * when HAVE_NEURON=1, will still return 0 but leave 
+ * hmem_support_status[FI_HMEM_NEURON].initialized and 
+ * hmem_support_status[FI_HMEM_NEURON].p2p_supported as false.
+ */
+void test_efa_hmem_support_status_update_neuron()
+{
+        int ret;
+        struct efa_resource resource = {0};
+        struct efa_domain *efa_domain;
+        uint32_t efa_device_caps_orig;
+
+        resource.hints = efa_unit_test_alloc_hints(FI_EP_RDM);
+        assert_non_null(resource.hints);
+
+        ret = fi_getinfo(FI_VERSION(1, 14), NULL, NULL, 0ULL, resource.hints, &resource.info);
+        assert_int_equal(ret, 0);
+
+        ret = fi_fabric(resource.info->fabric_attr, &resource.fabric, NULL);
+        assert_int_equal(ret, 0);
+
+        hmem_ops[FI_HMEM_NEURON].initialized = true;
```

Comment:
Clarify: in the current revision we are calling `fi_xxx` to initialize hints, fabric, etc., and we cannot do this instead:
```c
...
hmem_ops[FI_HMEM_NEURON].initialized = true;
efa_unit_test_resource_construct(&resource, FI_EP_RDM);
...
```
because we need this before `fi_domain` call?
```c
efa_device_caps_orig = g_device_list[0].device_caps;
g_device_list[0].device_caps |= EFADV_DEVICE_ATTR_CAPS_RDMA_READ;
```

Reply:
Correct, we need to modify these global variables after `fi_fabric` and before `fi_domain`, so we cannot call `efa_unit_test_resource_construct`